webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala

```diff
@@ -75,19 +78,23 @@ final case class GroupRestService(
 
   def putGroupStatus(iri: GroupIri, request: GroupStatusUpdateRequest, user: User): Task[GroupGetResponseADM] =
     for {
-      _ <- auth.ensureSystemAdminOrProjectAdminOfGroup(user, iri)
-      groupToUpdate <- groupService
-                         .findById(iri)
-                         .someOrFail(NotFoundException(s"Group <${iri.value}> not found."))
-      internal <- groupService.updateGroupStatus(groupToUpdate, request.status).map(GroupGetResponseADM.apply)
+      groupAndProject <- auth.ensureSystemAdminOrProjectAdminOfGroup(user, iri)
+      (group, _)       = groupAndProject
+      internal <- knoraGroupService
+                    .updateGroupStatus(group, request.status)
+                    .flatMap(groupService.toGroup)
+                    .map(GroupGetResponseADM.apply)
```

Comment:
nitpick: not sure if it makes sense to have for-comprehensions if we use map and flatmap in them; or vice versa, if it makes sense to use map and flatmap, if we're in a for-comprehension

Reply:
IMO this still reads better than chaining it all in one go. These lines could be also extracted to separate private methods, especially they are reused in `deleteGroup` methods below.

Reply:
1. I am totally ok to mix for comprehension with a flatmap here and there,
2. I am totally ok with a chain of flatmaps here and there
4. I had my reason of why the internal has become its own name and the intermediate transformative results where in a chain of map/flatMap for my personal preference of making it clear that toGroup and Foo.apply are simply transformations of intermediate unimportant results.
5. I have extracted the code duplication in a separate method and introduced intermediate result name due to popular demand. [b247906](https://github.com/dasch-swiss/dsp-api/pull/3204/commits/b247906d4777dc9fe9da175e9d0ac1c5aaefd5f7)

TBH, I find this kind of discussion exhausting and have a different preference for readability.

Reply:
It does look better with the flatMap refactored.